src/android/MifareUltralight.java

```diff
@@ -22,11 +22,11 @@ public void disconnect() throws Exception {
         mifare = null;
     }
 
-    public boolean unlockWithPin(int pin) throws Exception {
-        final byte[] pinAsByteArray = intToByteArray(pin);
+    public boolean unlockWithPin(long pin) throws Exception {
+        final byte[] pinAsByteArray = longToByteArray(pin);
         byte[] response = mifare.transceive(new byte[]{
                 (byte) 0x1B, // PWD_AUTH
-                pinAsByteArray[0], pinAsByteArray[1], pinAsByteArray[2], pinAsByteArray[3]
+                pinAsByteArray[4], pinAsByteArray[5], pinAsByteArray[6], pinAsByteArray[7]
```

Comment:
Probably I'm just missing something here but why we are sending only half of the pin here? 🤔

Reply:
The max size of the PIN is 0xFFFFFFFF, so the 4 first bytes would be ignored by the chip !

Reply:
Hmm.. I think I get what you mean so that should be fine.